eng/_core/cmd/build/build.go

```diff
@@ -241,15 +241,43 @@ func runCmd(cmd *exec.Cmd) error {
 	return cmd.Run()
 }
 
+func retry(attempts int, f func() error) error {
+	var i = 0
+	for ; i < attempts; i++ {
+		if attempts > 1 {
+			fmt.Printf("---- Running attempt %v of %v...\n", i+1, attempts)
+		}
+		err := f()
+		if err != nil {
+			if i+1 < attempts {
+				fmt.Printf("---- Attempt failed with error: %v\n", err)
+				continue
+			}
+			fmt.Printf("---- Final attempt failed.\n")
+			return err
+		}
+		break
+	}
+	fmt.Printf("---- Successful on attempt %v of %v.\n", i+1, attempts)
+	return nil
+}
+
 func getMaxMakeRetryAttempts() (int, error) {
-	const retryEnvVarName = "GO_MAKE_MAX_RETRY_ATTEMPTS"
-	a := os.Getenv(retryEnvVarName)
+	return getEnvIntOrDefault("GO_MAKE_MAX_RETRY_ATTEMPTS", 1)
+}
+
+func getMaxTestRetryAttempts() (int, error) {
+	return getEnvIntOrDefault("GO_TEST_MAX_RETRY_ATTEMPTS", 1)
+}
```

Comment:
IMO `getMaxMakeRetryAttempts` and `getMaxTestRetryAttempts` don't add much value here, I would rather directly call `getEnvIntOrDefault`. This way readers can directly see from where the number of attempts is taken from instead of jumping around.

Reply:
Totally agree. I just didn't give these funcs a second thought after duplicating `getMaxMakeRetryAttempts` and extracting the common bits.

Reply:
I'm also moving this into `main` so the command validates the env vars before anything happens, rather than e.g. waiting until after the build's done to tell someone they specified `GO_TEST_MAX_RETRY_ATTEMPTS` wrong.